x/ibc/20-transfer/module.go

```diff
@@ -119,3 +129,133 @@ func (am AppModule) BeginBlock(ctx sdk.Context, req abci.RequestBeginBlock) {
 func (am AppModule) EndBlock(ctx sdk.Context, req abci.RequestEndBlock) []abci.ValidatorUpdate {
 	return []abci.ValidatorUpdate{}
 }
+
+// Implement IBCModule callbacks
+func (am AppModule) OnChanOpenInit(
+	ctx sdk.Context,
+	order channelexported.Order,
+	connectionHops []string,
+	portID string,
+	channelID string,
+	chanCap capability.Capability,
+	counterparty channeltypes.Counterparty,
+	version string,
+) error {
+	// TODO: Enforce ordering, currently relayers use ORDERED channels
+
+	if counterparty.PortID != types.PortID {
+		return sdkerrors.Wrapf(porttypes.ErrInvalidPort, "counterparty has invalid portid. expected: %s, got %s", types.PortID, counterparty.PortID)
+	}
+
+	if version != types.Version {
+		return sdkerrors.Wrapf(porttypes.ErrInvalidPort, "invalid version: %s, expected %s", version, "ics20-1")
+	}
+
+	// Claim channel capability passed back by IBC module
+	if err := am.keeper.ClaimCapability(ctx, chanCap, ibctypes.ChannelCapabilityPath(portID, channelID)); err != nil {
+		return sdkerrors.Wrap(channel.ErrChannelCapabilityNotFound, err.Error())
+	}
+
+	// TODO: escrow
```

Comment:
Here is where we want to create the escrow address

Reply:
Looks like the current implementation just calculates the escrow address from portID, and channelID and uses the bank keeper to send and receive money. This seems like it should work?

Should I make these escrow addresses module accounts instead?

https://github.com/cosmos/cosmos-sdk/blob/ibc-alpha/x/ibc/20-transfer/keeper/relay.go#L97

https://github.com/cosmos/cosmos-sdk/blob/ibc-alpha/x/ibc/20-transfer/keeper/relay.go#L186

Reply:
This is OK for now, I suppose, maybe they should be module accounts but that can be a follow-up.